client.go

```diff
@@ -131,17 +131,15 @@ func (c *client) deleteServicePrincipal(ctx context.Context, spObjectID string,
 }
 
 // assignRoles assigns Azure roles to a service principal.
-func (c *client) assignRoles(ctx context.Context, spID string, roles []*AzureRole) ([]string, error) {
+func (c *client) assignRoles(ctx context.Context, spID string, roles []*AzureRole, assignmentIDs []string) ([]string, error) {
 	var ids []string
 
-	for _, role := range roles {
-		assignmentID, err := uuid.GenerateUUID()
-		if err != nil {
-			return nil, err
-		}
-
+	for i, role := range roles {
```

Comment:
I'm thinking we should return an error if `len(roles) != len(assignmentIDs)` above this loop. That will ensure that we can safely use `assignmentIDs[i]` within the loop. The lengths are equal with our current usage of `assignRoles()`, but by adding the check, we avoid a panic if someone misuses this in the future.